Algorithms/ValidParentheses.cs

```diff
@@ -22,22 +22,38 @@ public class ValidParentheses
 			charsOfClose.Add('[', ']');
 			int lengthString = stringCharacters.Length, i = 0;
 
-			while (i < lengthString)
-			{
-				if (stringCharacters[i] == '(' || stringCharacters[i] == '{' || stringCharacters[i] == '[')
+			while (i < lengthString) {
+				if (validOpenChars(stringCharacters, i))
 					characters.Push(stringCharacters[i]);
-				else if (characters.Count > 0)
-				{
+
+				else if (characters.Count > 0) {
 					if (charsOfClose[characters.Peek()] == stringCharacters[i])
 						characters.Pop();
 					else
 						return false;
-				}
-				else
+				} else
 					return false;
 				i++;
 			}
-			return characters.Count == 0 ? true : false;
+			return characters.Count == 0 & true;
+		}
+		/// <summary>
+		/// Valid open chars
```

Comment:
If you are going to include comments, make them more useful than just the name of the function